### DISCUSSION TITLE
Sharing Files from within a protected folder

### DISCUSSION THREAD

**Discussion by angsep** [spencerwooo/onedrive-vercel-index#337] [Q&A] [answered]
Can I protect the folder but still share the files in the folder using the raw=true url parameter without using a password?

**Comment by spencerwooo** [comment]
Nope, if you password protect a folder, then the files inside are non-shareable with raw links.

**Reply by hirockstar** [reply]
#337 @spencerwooo bro make it possible please. it's good to have.

**Reply by spencerwooo** [reply]
What's the purpose of password protecting a folder then sharing its contents? I can't find any justifications - so no, not going to support this.

**Reply by depiuri** [reply]
A comment without scope to offend. I am very grateful for your project. Imagine that I want to share a file but I want privacy so that the user does not see the other files. I just want him to be able to access via download just that one file. It's a safe way to directly share files without fear of them having access to our entire cloud.

**Comment by spencerwooo** [comment]
@ChikoritaA I thought I'd reply to u here in a new thread.

So, first of all, this project is not for encrypting / protecting your files on OneDrive. It's for **presenting** them in a nicer way than the default OneDrive interface.

Additionally, the _protected route_ feature is a coarse and unreliable implementation that is _not_ using **third-party battle-tested user identity validation libraries**. Rather, it just consists of a few basic lines of code that I wrote with no security guarantee.

Finally, OneDrive does provide you with a way to share the direct links of your files, and also manges permissions respectively (correct me if I'm wrong).

So, I would say that this is outside of this project's scope, and any further discussion is unnecessary. Thanks for understanding.

**Comment by spencerwooo** [comment, marked as answer]
All right, it's supported. Check docs: [Caveats - Password protected folders](https://ovi.swo.moe/docs/features/protected-folders#caveats). ~~You guys happy now? :)~~

> Raw file link redirect API accepts the URL parameter `&odpt=<hashed_token>` for password authentication.
